src/main/java/org/testng/TestRunner.java

```diff
@@ -862,6 +866,16 @@ private void privateRun(XmlTest xmlTest) {
     return result;
   }
 
+  private Map<String, String> getParameters(IMethodInstance im) {
```

Comment:
Could be static if the `XmlTest` comes from `im.getMethod().getXmlTest()`.

And what about `im.getMethod().findMethodParameters(im.getMethod().getXmlTest())`?